api/v1alpha1/cdapmaster_types.go

```diff
@@ -184,8 +184,12 @@ type CDAPExternalServiceSpec struct {
 	CDAPScalableServiceSpec `json:",inline"`
 	// ServiceType is the service type in kubernetes, default is NodePort.
 	ServiceType *string `json:"serviceType,omitempty"`
+	// LoadBalancerIP is the static IP address reserved for the service's load balancer.
```

Comment:
Do we need to use preallocated IP? Or can it be auto allocated when the underlying ILB is created?

Reply:
This is an optional field, so it can be skipped. If cloud provider supports, we can provide a static IP. This is useful if we want to create VPC routes before the LB is created or we want to ensure the routes remain correct even if the load balancer is re-created.
* [k8s docs](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer:~:text=Some%20cloud%20providers%20allow%20you%20to%20specify%20the%20loadBalancerIP)
* [GKE docs](https://cloud.google.com/kubernetes-engine/docs/concepts/service-load-balancer-parameters#spd-static-ip)

Reply:
On thinking a bit, this field isn't required from CDF perspective. Should I remove it?

Reply:
Discussed offline, it is file to keep the field even though it is not a requirement for CDF usage.